react/admin/ReturnDetails/ReturnDetailsContainer.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react'
+import {
+  Layout,
+  PageHeader,
+  PageBlock,
+  Spinner,
+  EmptyState,
+} from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import { useRuntime } from 'vtex.render-runtime'
+
+import { UpdateRequestStatus } from './components/UpdateRequestStatus'
+import { useReturnDetails } from '../hooks/useReturnDetails'
+import { VerifyItemsPage } from './components/VerifyItems/VerifyItemsPage'
+
+type Pages = 'return-details' | 'verify-items'
+
+export const ReturnDetailsContainer = () => {
+  const [detailsPage, setDetailsPage] = useState<Pages>('return-details')
+  const { data, error, loading } = useReturnDetails()
+
+  const { navigate } = useRuntime()
+
+  const handleViewVerifyItems = (page: Pages) => {
+    setDetailsPage(page)
+  }
+
+  return (
+    <Layout
+      fullWidth
+      pageHeader={
+        <PageHeader
+          title={
+            <FormattedMessage id="admin/return-app.return-request-details.page-header.title" />
+          }
+          linkLabel={
+            <FormattedMessage id="admin/return-app.return-request-details.page-header.link-label" />
+          }
+          onLinkClick={() => {
+            navigate({
+              to: '/admin/returns/requests',
+            })
+          }}
+        />
+      }
+    >
+      <PageBlock variation="full" fit="fill">
+        {error ? (
+          <EmptyState
+            title={
+              <FormattedMessage id="admin/return-app.return-request-details.error.title" />
+            }
+          >
+            <FormattedMessage id="admin/return-app.return-request-details.error.description" />
+          </EmptyState>
+        ) : loading ? (
+          <Spinner />
+        ) : !data?.returnRequestDetails ? null : (
```

Comment:
What do you think of making the returned components, at least from `data`, more readable? 
Maybe having an additional `wrapper` component for the Layout, or a functional component that accepts the props required for either page.

Reply:
Yeah. Logic was a little bit hard to follow. Sorry. I added a Loader component. Let me know what you think now.